From 3a1cb43a09c9d7902c3129e1bc6a622443473365 Mon Sep 17 00:00:00 2001
From: Colin Patrick Mccabe <cmccabe@cloudera.com>
Date: Thu, 18 Jun 2015 11:30:29 -0700
Subject: [PATCH 1189/2848] HADOOP-12100. ImmutableFsPermission should not
 override applyUmask since that method doesn't
 modify the FsPermission (Bibin A. Chundatt via
 Colin P. McCabe)

(cherry picked from commit 6e0a9f92fe0052d39b95a605174b3f6423c6aae7)
(cherry picked from commit 2946e92f79a73cccf9533f31ad40471046496cb2)
(cherry picked from commit e5f9703419234123ef266fcee05cd32eea8d9b79)

Change-Id: I7720682dbbd06ab42f614a5d329e002d5404cd9c
---
 .../apache/hadoop/fs/permission/FsPermission.java  |    7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/permission/FsPermission.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/permission/FsPermission.java
index 264a095..0258293 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/permission/FsPermission.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/permission/FsPermission.java
@@ -375,13 +375,10 @@ else if (unixSymbolicPermission.length() != MAX_PERMISSION_LENGTH) {
     public ImmutableFsPermission(short permission) {
       super(permission);
     }
-    @Override
-    public FsPermission applyUMask(FsPermission umask) {
-      throw new UnsupportedOperationException();
-    }
+
     @Override
     public void readFields(DataInput in) throws IOException {
       throw new UnsupportedOperationException();
-    }    
+    }
   }
 }
-- 
1.7.9.5

